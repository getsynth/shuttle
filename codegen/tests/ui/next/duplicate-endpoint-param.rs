shuttle_codegen::app! {
    #[shuttle_codegen::endpoint(method = get, method = get)]
    async fn hello() -> &'static str {
        "Hello, World!"
    }

    #[shuttle_codegen::endpoint(route = "/goodbye", route = "/goodbye")]
    async fn goodbye() -> &'static str {
        "Goodbye, World!"
    }
}
