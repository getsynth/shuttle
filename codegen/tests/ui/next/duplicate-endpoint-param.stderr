error: duplicate endpoint method

         = help: The endpoint `method` should only be set once.

 --> tests/ui/next/duplicate-endpoint-param.rs:2:47
  |
2 |     #[shuttle_codegen::endpoint(method = get, method = get)]
  |                                               ^^^^^^

error: no route provided

         = help: Add a route to your endpoint: `route = "/hello")`

 --> tests/ui/next/duplicate-endpoint-param.rs:2:32
  |
2 |     #[shuttle_codegen::endpoint(method = get, method = get)]
  |                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: duplicate endpoint route

         = help: The endpoint `route` should only be set once.

 --> tests/ui/next/duplicate-endpoint-param.rs:7:53
  |
7 |     #[shuttle_codegen::endpoint(route = "/goodbye", route = "/goodbye")]
  |                                                     ^^^^^

error: no method provided

         = help: Add a method to your endpoint: `method = get`

 --> tests/ui/next/duplicate-endpoint-param.rs:7:32
  |
7 |     #[shuttle_codegen::endpoint(route = "/goodbye", route = "/goodbye")]
  |                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0601]: `main` function not found in crate `$CRATE`
  --> tests/ui/next/duplicate-endpoint-param.rs:11:2
   |
11 | }
   |  ^ consider adding a `main` function to `$DIR/tests/ui/next/duplicate-endpoint-param.rs`
